src/dune_rules/melange/melange_rules.ml

```diff
@@ -333,14 +338,19 @@ let setup_runtime_assets_rules sctx ~dir ~target_dir ~mode
         Super_context.add_rule ~loc ~dir ~mode sctx
           (Action_builder.copy ~src ~dst))
   and+ () =
+    let deps =
+      Action_builder.paths
+        (non_copy @ List.rev_map copy ~f:(fun (_, target) -> Path.build target))
+    in
+    let* () =
+      let alias =
+        Alias.make Melange_stanzas.Emit.implicit_alias ~dir:target_dir
+      in
+      Rules.Produce.Alias.add_deps alias deps
+    in
     match mel.alias with
```

Comment:
You can run these in parallel
```suggestion
    let+ () =
      let alias =
        Alias.make Melange_stanzas.Emit.implicit_alias ~dir:target_dir
      in
      Rules.Produce.Alias.add_deps alias deps
    and+ () =
    match mel.alias with
```

Reply:
Done in https://github.com/ocaml/dune/pull/7327/commits/950d3f91a7f6231d4e49de706e46b339dede5389.